hosttest/convolutiontest.c

```diff
@@ -0,0 +1,52 @@
+#include <math.h>
+#include <stdbool.h>
+#include <stdio.h>
+
+#include "convolution.h"
+
+#define ARRAY_LEN(_a)	(sizeof(_a)/sizeof(_a[0]))
+#define EPSILON 0.01f
+
+float in[] = {1, 2, 3, 4};
+float filter[] = {-1, 5, 3};
+float out[10];
+
+float expected[10] = {-1.0f, 3.0f, 10.0f, 17.0f, 29.0f, 12.0f, 0.0f, };
```

Comment:
I get an incorrect answer ("FAILED") with these, but if `in` and `filter` are switched, I get the correct answer.

Since convolution is supposed to be commutative, switching the `in` and `filter` shouldn't have any effect.  My guess (and what I was looking for with this test) was issues when size(in) < size(filter) or vice versa.